pypeit/tests/test_traceslits.py

```diff
@@ -24,42 +24,47 @@ def chk_for_files(root):
 
 @dev_suite_required
 def test_addrm_slit():
+    # TODO This TEST needs to be rewritten becuase from_master_files is defunct. THe master only includes
+    # the master file and not the entire history of the code as before.
+
     # Check for files
     mstrace_root = os.path.join(os.getenv('PYPEIT_DEV'), 'Cooked', 'Trace',
                                 'MasterTrace_KeckLRISr_400_8500_det1')
     assert chk_for_files(mstrace_root)
     # Load
-    traceSlits = traceslits.TraceSlits.from_master_files(mstrace_root)
-    norig = traceSlits.nslit
+    #traceSlits = traceslits.TraceSlits.from_master_files(mstrace_root)
+    #norig = traceSlits.nslit
 
     #  Add dummy slit
     #  y_spec, left edge, right edge on image
-    add_user_slits = [[1024, 140, 200]]
-    traceSlits.lcen, traceSlits.rcen = trace_slits.add_user_edges(
-        traceSlits.lcen, traceSlits.rcen, add_user_slits)
+    #add_user_slits = [[1024, 140, 200]]
+    #traceSlits.lcen, traceSlits.rcen = trace_slits.add_user_edges(
+    #    traceSlits.lcen, traceSlits.rcen, add_user_slits)
     # Test
-    assert traceSlits.nslit == (norig+1)
-    xcen0 = np.median((traceSlits.lcen[:,0] + traceSlits.rcen[:,0])/2.)
-    assert np.abs(xcen0-170) < 3
+    #assert traceSlits.nslit == (norig+1)
+    #xcen0 = np.median((traceSlits.lcen[:,0] + traceSlits.rcen[:,0])/2.)
+    #assert np.abs(xcen0-170) < 3
 
     # Remove it
-    rm_user_slits = [[1024, 170]]
-    traceSlits.lcen, traceSlits.rcen = trace_slits.rm_user_edges(
-        traceSlits.lcen, traceSlits.rcen, rm_user_slits)
-    assert traceSlits.nslit == norig
+    #rm_user_slits = [[1024, 170]]
+    #traceSlits.lcen, traceSlits.rcen = trace_slits.rm_user_edges(
+    #    traceSlits.lcen, traceSlits.rcen, rm_user_slits)
+    #assert traceSlits.nslit == norig
 
 @dev_suite_required
 def test_chk_kast_slits():
     # Red, blue
     for root in ['MasterTrace_ShaneKastred_600_7500_d55', 'MasterTrace_ShaneKastblue_600_4310_d55']:
         mstrace_root = os.path.join(os.getenv('PYPEIT_DEV'), 'Cooked', 'Trace', root)
         assert chk_for_files(mstrace_root)
-        traceSlits = traceslits.TraceSlits.from_master_files(mstrace_root)
-        norig = traceSlits.nslit
+        # TODO This TEST needs to be rewritten becuase from_master_files is defunct. THe master only includes
```

Comment:
`from_master_files` isn't needed.  Just need to instantiate the class, 
load the `mstrace` image and `run()`

Reply:
There is no way to reinstantiate the class from file currently, since not all the tags are saved. I  strongly disagree that this is an intelligent way to program. If you want to code in that way go ahead but I will not maintain it or any tests based on it. The point is PypeIt does not ever need to reinstantiate a class form a file at runtime. The slit tracing runs on images that are passed as an argument at runtime. You should be testing that functionality and not this in my opinion silly reinsantiation which is never done anywhere except in this test.

Reply:
Which tags are not saved that are needed for instantiation?